tekton/pipeline_run.go

```diff
@@ -135,6 +138,34 @@ func (r *ReleasePipelineRun) WithEnterpriseContractPolicy(enterpriseContractPoli
 	return r
 }
 
+// copyPipelinesAsCodeMeta copies key/value pairs in src adding them to dst.
+// When a key in src is already present in dst, no copy happens.
+// When the src key does not contain the prefix string, no copy happens.
+func copyPipelinesAsCodeMeta(src, dst map[string]string) {
+	for key, val := range src {
+		if _, err := dst[key]; !err && strings.HasPrefix(key, pipelinesAsCodeMetaPrefix) {
+			dst[key] = val
+		}
+	}
+}
+
+// WithPipelinesAsCodeMeta conditionally copies Labels and Annotations from Release to PipelineRun
+// When Annotations or Labels maps do not exist, then they are created.
+// Only things who match pipelinesAsCodeMetaPrefix are allowed to be copied.
+func (pipelineRun *ReleasePipelineRun) WithPipelinesAsCodeMeta(release *v1alpha1.Release) *ReleasePipelineRun {
```

Comment:
Please, be consistent with receiver names.

```suggestion
func (r *ReleasePipelineRun) WithPipelinesAsCodeMeta(release *v1alpha1.Release) *ReleasePipelineRun {
```

Reply:
You are setting Annotations here so probably the `WithReleaseAndApplicationLabels` function should be renamed to `WithReleaseAndApplicationMetadata`.

Reply:
This function was deleted, and the code moved into `WithReleaseAndApplicationMetadata`. PTAL